src/containers/profile/profile.component.jsx

```diff
@@ -9,19 +11,25 @@ const Profile = (props) => {
   const userFromState = useSelector((state) => state.users.user);
 
   const updateUserInfo = () => {
-    if (!username || !imageAsFile) {
-      alert("Please enter a username and upload an avatar");
-    } else {
-      const userWithUsername = {
-        id: auth.currentUser.uid,
-        username: username,
-        user: userFromState,
-      };
-      addUsernameToStateThunk(userWithUsername);
-      uploadProfileAvatarThunk({ image: imageAsFile, username });
+    usernameCheck(username).then((res) => {
```

Comment:
username check not a very clear function name. Best add comment above the function to explain what the function does. I think there are some examples of this in the codebase